specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2021-10-01-preview/scanningService.json

```diff
@@ -486,6 +486,181 @@
         }
       }
     },
+    "/credentials/{credentialName}": {
+      "get": {
+        "tags": [
+          "Credential"
+        ],
+        "description": "Gets credential information",
+        "operationId": "Credential_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "credentialName",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Credential"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Scanning Service",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Credential_Get": {
+            "$ref": "./examples/Credential_Get.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Credential"
+        ],
+        "description": "Creates an instance of a credential",
+        "operationId": "Credential_CreateOrUpdate",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "credentialName",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Credential"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
You should use status 201 if the resource was created, and 200 if it was replaced.

Reply:
Added logic to return 201 on create and 200 on update.